x-pack/plugins/alerting/public/pages/maintenance_windows/components/create_maintenance_windows_form.tsx

```diff
@@ -109,6 +115,32 @@ export const CreateMaintenanceWindowForm = React.memo<CreateMaintenanceWindowFor
     const isRecurring = recurring || false;
     const showTimezone = isBrowser || initialValue?.timezone !== undefined;
 
+    const closeModal = () => setIsModalVisible(false);
+    const showModal = () => setIsModalVisible(true);
+
+    let modal;
+    if (isModalVisible) {
+      modal = (
+        <EuiConfirmModal
+          title={i18n.ARCHIVE_TITLE}
+          onCancel={closeModal}
+          onConfirm={() => {
+            closeModal();
+            archiveMaintenanceWindow(
+              { maintenanceWindowId: maintenanceWindowId!, archive: true },
+              { onSuccess }
+            );
+          }}
+          cancelButtonText={i18n.CANCEL}
+          confirmButtonText={i18n.ARCHIVE_TITLE}
+          defaultFocusedButton="confirm"
+          buttonColor="danger"
+        >
+          <p>{i18n.ARCHIVE_CALLOUT_SUBTITLE}</p>
+        </EuiConfirmModal>
+      );
+    }
```

Comment:
nitpick: useMemo or move to it's own component

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/155201/commits/9d6e6730e18bd54c5bece8e97a0c801d232edd3c